custom/icds_reports/migrations/0104_citus_composite_key.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.conf import settings
+from corehq.sql_db.operations import HqRunPython
+
+
+def _citus_composite_key_migration(apps, schema_editor):
+    models_to_update = [
+        'aggregateawcinfrastructureforms',
+        'aggregatebirthpreparednesforms',
+        'aggregateccsrecorddeliveryforms',
+        'aggregateccsrecordpostnatalcareforms',
+        'aggregateccsrecordthrforms',
+        'aggregatechildhealthdailyfeedingforms',
+        'aggregatechildhealthpostnatalcareforms',
+        'aggregatechildhealththrforms',
+        'aggregatecomplementaryfeedingforms',
+        'aggregategrowthmonitoringforms',
+        'dailyattendance',
+        'awwincentivereport'
+    ]
+    if not getattr(settings, 'IS_ON_CITUSDB_BACKEND', False):
+        return
+    else:
+        with schema_editor.connection.cursor() as cursor:
+            for model in models_to_update:
+                model_cls = apps.get_model('icds_reports', model)
+                pkey = '{}_pkey'.format(model_cls._meta.db_table)
+                fields = list(model_cls._meta.unique_together)[0]
+                columns = [model_cls._meta.get_field(field).column for field in fields]
+                index = schema_editor._create_index_name(model_cls, columns, suffix="_uniq")
+                cursor.execute("""
+                    CREATE UNIQUE INDEX {index} on "{table}" ({cols});
+                    ALTER TABLE "{table}" DROP CONSTRAINT {pkey},
+                    ADD CONSTRAINT {pkey} PRIMARY KEY USING INDEX {index};
+                """.format(
+                    table=model_cls._meta.db_table,
+                    pkey=pkey,
+                    index=index,
+                    cols=','.join(columns)
+                ))
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('accounting', '0103_drop_update_aggregate_awc_data'),
+    ]
+
+    operations = [
+        HqRunPython(_citus_composite_key_migration),
```

Comment:
This will need corresponding Django state operations otherwise Django will think the indexes haven't been added. Probably the best route would be to make the function above work for a single table and then you can have something like this:
```
operations = [
    migrations.RunSQL(get_composite_index_sql('icds-reports', 'modelname'), state_operations=....)
]
```

Reply:
Since Django doesn't support composite key, how does an example state operation look like for a model?

Reply:
it would be the state operation to add the unique index which you've added to the django models